lib/rubocop/cop/naming/forbidden_variable_name.rb

```diff
@@ -0,0 +1,79 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Naming
+      # This cop makes sur that variables used are not fordidden for use.
+      # If a recommended variable is present it will display it.
+      #
+      # The recommendations are configured in the `Recommendations`
+      # of the `Naming/ForbiddenVariableName` cop.
+      #
+      # @example
+      #   # bad
+      #   def do_something_with_class(clazz)
+      #   end
+      #
+      #   # bad
+      #   classes.map { |clazz| puts clazz.name }
+      #
+      #   # bad
+      #   clazz = Array
+      #
+      #   # good
+      #   def do_something_with_class(klass)
+      #   end
+      #
+      #   # good
+      #   classes.map { |klass| puts klass.name }
+      #
+      #   # good
+      #   klass = Array
+      class ForbiddenVariableName < Cop
+        MSG = 'Use `%<good>s` instead of `%<bad>s`.'.freeze
+        MSG_NO_RECOMENDATION = 'Do not use `%<bad>s`.'.freeze
+
+        def on_lvasgn(node)
+          name, = *node
+          return unless name
+
+          check_name(node, name.to_s.downcase)
+        end
+        alias on_ivasgn    on_lvasgn
```

Comment:
Many of these aliases are not covered by tests, so please add spec examples with instance variables, constants, etc.